server/src/validation/items.validation.ts

```diff
@@ -58,4 +58,41 @@ export const validateAddItemRequest = () => {
                 throw new Error('categoryId does not exist');
             })
     ];
-};
\ No newline at end of file
+};
+
+export const validateEditItemRequest = () => {
+    return [
+        body('_id', 'Invalid itemId'),
+        body('name', 'Invalid name')
+            .isString()
+            .notEmpty({ ignore_whitespace: true }),
+        body('imageUrl', 'Invalid imageUrl')
+            .isString()
+            .notEmpty({ ignore_whitespace: true }),
+        body('categoryId', 'Invalid categoryId')
+            .isString()
+            .isString()
+            .custom(async (_id: string, { req }: { req: Request }) => {
+                const itemId: string = _id || req.body.itemId;
+                if (await ItemModel.findOne({ _id: itemId, userId: req.user.id })) {
```

Comment:
Are you sure req.user.id is working?
I think it should be `(req as AuthenticatedRequest).token.userId`
Same in `validateDeleteItemRequest`